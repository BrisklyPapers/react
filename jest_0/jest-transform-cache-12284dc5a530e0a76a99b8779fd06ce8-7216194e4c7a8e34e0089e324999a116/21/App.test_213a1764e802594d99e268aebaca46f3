'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var ReactShallowRenderer = require('react-test-renderer/shallow');

var mockStore = (0, _reduxMockStore2.default)();
var renderer = new ReactShallowRenderer();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = renderer.render(
        _react2.default.createElement(_App2.default, null));

        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});