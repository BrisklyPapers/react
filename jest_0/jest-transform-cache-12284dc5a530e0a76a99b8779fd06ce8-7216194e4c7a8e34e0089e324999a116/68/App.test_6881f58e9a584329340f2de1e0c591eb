'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/App', function () {
    it('renders correctly', function () {
        var app = _reactTestRenderer2.default.create(
        _react2.default.createElement(_App2.default, null));

        expect(app.toJSON()).toMatchSnapshot();
    });
});