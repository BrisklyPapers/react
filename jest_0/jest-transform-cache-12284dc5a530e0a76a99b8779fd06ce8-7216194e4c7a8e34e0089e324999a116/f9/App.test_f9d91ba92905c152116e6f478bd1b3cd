'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = _reactTestRenderer2.default.create(
        _react2.default.createElement(_MuiThemeProvider2.default, null,
            _react2.default.createElement(_reactRedux.Provider, { store: mockStore({ searchedText: "" }) },
                _react2.default.createElement(_App2.default, null))));



        expect(app.toJSON()).toMatchSnapshot();
    });
});