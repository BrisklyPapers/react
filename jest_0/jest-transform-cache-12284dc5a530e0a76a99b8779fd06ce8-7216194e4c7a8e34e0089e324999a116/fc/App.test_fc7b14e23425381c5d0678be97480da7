'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var ReactShallowRenderer = require('react-test-renderer/shallow');

var mockStore = (0, _reduxMockStore2.default)();
var renderer = new ReactShallowRenderer();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = renderer.render(
        _react2.default.createElement(_MuiThemeProvider2.default, null,
            _react2.default.createElement(_reactRedux.Provider, { store: mockStore({ searchedText: "" }) },
                _react2.default.createElement(_App2.default, null))));



        expect(app.getRenderOutput()).toMatchSnapshot();
    });
});