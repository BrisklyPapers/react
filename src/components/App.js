import React from 'react';
import Navigation from '../containers/Navigation';
import IndexPage from '../containers/IndexPage';
import ResultPage from '../containers/ResultPage';
import FileUploadPage from '../containers/FileUploadPage';
import SearchResults from '../containers/SearchResults';
import DropZone from '../containers/DropZone';
import IndexPageHead from './IndexPageHead';
import IndexPageDescription from './IndexPageDescription';
import Radium from 'radium';
import {connect} from 'react-redux';

class App extends React.Component {
    render() {
        return (
            <div>
                <div style={styles.divOnTop[this.props.cssStyle]}>
                    <Navigation />
                    <IndexPageHead />
                    <ResultPage>
                        <SearchResults />
                    </ResultPage>
                    <FileUploadPage>
                        <DropZone />
                    </FileUploadPage>
                </div>
                <IndexPageDescription />
            </div>
        )
    };
}

var styles = {
    divOnTop: {
        index: {
            backgroundImage: 'url("/home_background_1920cs.jpg")',
            height: "100vh",
            backgroundSize: 'cover',
            backgroundPosition: 'bottom center',
            backgroundRepeat: 'no-repeat',
            position: "relative"
        },
        login: {
            backgroundImage: 'url("/login_background_1920x420.jpg")',
            minHeight: "100vh",
            height: "100%",
            backgroundSize: '100% 1px',
            backgroundPosition: 'bottom center',
            backgroundRepeat: 'repeat-y',
            position: "relative"
        }
    }
};

export default Radium(App);