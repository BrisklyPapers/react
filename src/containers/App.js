import { connect } from 'react-redux';
import AppComponent from '../components/App';

const mapStateToProps = (state) => {
    return {
        loggedIn: state.logged_in,
        cssStyle: state.logged_in ? 'login' : 'index'
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AppComponent);