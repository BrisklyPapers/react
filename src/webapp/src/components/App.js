import React from 'react';
import Search from '../containers/Search';
import SearchResults from '../containers/SearchResults';

const App = () => (
    <div>
        <Search />
        <SearchResults />
    </div>
);

export default App;