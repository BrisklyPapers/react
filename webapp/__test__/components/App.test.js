import React from 'react';
import App from '../../src/components/App.js';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Provider } from 'react-redux';
import fetchMock from 'fetch-mock'
import configureMockStore from 'redux-mock-store'

const mockStore = configureMockStore();

describe('components/App', () => {
    it('renders correctly', () => {
        const app = renderer.create(
            <MuiThemeProvider>
                <Provider store={mockStore({searchedText: ""})}>
                    <App />
                </Provider>
            </MuiThemeProvider>
        );
        expect(app.toJSON()).toMatchSnapshot();
    });
});