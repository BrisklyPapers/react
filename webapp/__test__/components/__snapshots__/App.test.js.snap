// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "white",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 56,
          "justifyContent": "space-between",
          "muiPrepared": true,
          "padding": "0px 24px",
        }
      }
    >
      <div
        className={undefined}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginLeft": -24,
            "marginRight": undefined,
            "muiPrepared": true,
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.4)",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 20,
              "lineHeight": "56px",
              "muiPrepared": true,
              "overflow": "hidden",
              "paddingRight": 16,
              "position": "relative",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Briskly Papers
        </span>
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 48,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 256,
            }
          }
        >
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 1,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            Search here ..
          </div>
          <input
            disabled={false}
            id="search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <button
          disabled={false}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "#ff4081",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": "10px 24px",
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "muiPrepared": true,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "verticalAlign": "middle",
                }
              }
            >
              Upload files
            </span>
          </div>
        </button>
      </div>
      <div
        className={undefined}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginLeft": undefined,
            "marginRight": -24,
            "muiPrepared": true,
            "position": "relative",
          }
        }
      >
        <button
          disabled={true}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.3)",
              "cursor": "default",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": "10px 24px",
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Login
          </span>
        </button>
      </div>
    </div>
  </div>
  <div />
  <div />
</div>
`;
